test/jb_projects/claim_handle.test.js

```diff
@@ -0,0 +1,290 @@
+        METADATA_CID,
+      );
+
+    await jbProjectsStore
+      .connect(projectOwner)
+      .transferHandleOf(
+        PROJECT_ID_1,
+        deployer.address,
+        ethers.utils.formatBytes32String(PROJECT_HANDLE_3),
+      );
+
+    let tx = await jbProjectsStore
+      .connect(projectOwner)
+      .challengeHandle(ethers.utils.formatBytes32String(PROJECT_HANDLE_2));
+
+    await fastForward(tx.blockNumber, ethers.BigNumber.from(12536000));
+
+    await expect(
+      jbProjectsStore
+        .connect(deployer)
+        .claimHandle(
+          ethers.utils.formatBytes32String(PROJECT_HANDLE_2),
+          deployer.address,
+          PROJECT_ID_2,
+        ),
+    ).to.be.revertedWith('0x0c: UNAUTHORIZED');
+  });
+
+  it(`Can't claim handle and assign to inexistent or not owned projectId`, async function () {
+    const { projectOwner, deployer, addrs, jbProjectsStore, mockJbOperatorStore } = await setup();
+
+    await jbProjectsStore
+      .connect(deployer)
+      .createFor(
+        projectOwner.address,
+        ethers.utils.formatBytes32String(PROJECT_HANDLE_1),
+        METADATA_CID,
+      );
+
+    let handleReceiver = addrs[0];
+    await jbProjectsStore
+      .connect(projectOwner)
+      .transferHandleOf(
+        PROJECT_ID_1,
+        handleReceiver.address,
+        ethers.utils.formatBytes32String(PROJECT_HANDLE_2),
+      );
+
+    await mockJbOperatorStore.mock.hasPermission
+      .withArgs(
+        handleReceiver.address,
+        deployer.address,
+        PROJECT_ID_1,
+        CLAIM_HANDLE_PERMISSION_INDEX,
+      )
+      .returns(true);
+
+    await expect(
+      jbProjectsStore
+        .connect(handleReceiver)
+        .claimHandle(
+          ethers.utils.formatBytes32String(PROJECT_HANDLE_1),
+          projectOwner.address,
+          PROJECT_ID_2,
+        ),
+    ).to.be.reverted;
+  });
+
+  it(`Can't claim handle even if its recipient who does it but without permissions`, async function () {
+    const { projectOwner, deployer, addrs, jbProjectsStore, mockJbOperatorStore } = await setup();
+
+    await jbProjectsStore
+      .connect(deployer)
+      .createFor(
+        projectOwner.address,
+        ethers.utils.formatBytes32String(PROJECT_HANDLE_1),
+        METADATA_CID,
+      );
+
+    let handleReceiver = addrs[0];
+    await jbProjectsStore
+      .connect(projectOwner)
+      .transferHandleOf(
+        PROJECT_ID_1,
+        handleReceiver.address,
+        ethers.utils.formatBytes32String(PROJECT_HANDLE_2),
+      );
+
+    await mockJbOperatorStore.mock.hasPermission
+      .withArgs(
+        handleReceiver.address,
+        deployer.address,
+        PROJECT_ID_1,
+        CLAIM_HANDLE_PERMISSION_INDEX,
+      )
+      .returns(false);
+
+    await expect(
+      jbProjectsStore
+        .connect(handleReceiver)
+        .claimHandle(
+          ethers.utils.formatBytes32String(PROJECT_HANDLE_1),
+          projectOwner.address,
+          PROJECT_ID_2,
```

Comment:
this is failing because handleReceiver isn't the owner of PROJECT_ID_2

Reply:
Correct, this is fixed.